functions/index.js

```diff
@@ -169,11 +169,35 @@ function handleError(req, res, payload = {}, step, incomingError, shouldExit = t
     incomingError = new Error(incomingError);
   }
 
+  // Handle Firebase Messaging errors with appropriate status codes
+  if (incomingError.code && incomingError.code.startsWith('messaging/')) {
+    const errorCode = incomingError.code.replace('messaging/', '');
+
+    // For specific token errors, skip reporting and return immediately
+    if (
+      errorCode === 'invalid-registration-token' ||
+      errorCode === 'registration-token-not-registered'
+    ) {
+      if (!shouldExit) {
+        return true;
+      }
+
+      return res.status(404).send({
```

Comment:
404 for invalid token does not seem right, should this not be (401|403)?

Reply:
True, I don't want to find out if the client is going to break in a new way with a different status code. Changing back to 500.